src/modules/blogs/blog.routes.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from "express";
+import { BlogController } from "./blog.controller";
+import { validateDto } from "../../common/validators";
+import { CreateBlogDto } from "./dto/create-blog.dto";
+import { authenticateToken } from "../../common/middlewares/authenticate-token";
+
+export const BlogRouter = Router();
+
+BlogRouter.post(
+  "/",
+  authenticateToken,
+  validateDto(CreateBlogDto),
+  BlogController.createBlogHandler,
+);
+
+BlogRouter.get("/", authenticateToken, BlogController.getMyBlogsHandler);
+BlogRouter.get("/public", BlogController.getPublicBlogsHandler);
+BlogRouter.get("/:id", authenticateToken, BlogController.getBlogByIdHandler);
```

Comment:
this must be public because without login, users can see posts. but only writing comment or putting likes must be protected.

Reply:
I just wanted to separate user's own blogs from all blogs....